sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseFacadeEjb.java

```diff
@@ -1774,6 +1774,34 @@ public void archiveOrDearchiveCase(String caseUuid, boolean archive) {
 		caseService.updateArchived(Collections.singletonList(caseUuid), archive);
 	}
 
+	/**
+	 * Handles potential changes of related tasks that needs to be done after
+	 * a case has been created/saved
+	 */
+	private void updateTasksOnCaseChanged(Case newCase, CaseDataDto existingCase) {
+		// In case the health facility of a case changes
+		// check if a new surveillance officer for this facility was also assigned.
+		// If yes re-assign the tasks associated with this case to this SurvOff
+		if (existingCase != null
+			&& newCase.getHealthFacility() != null
+			&& existingCase.getHealthFacility() != null
+			&& !newCase.getHealthFacility().getUuid().equals(existingCase.getHealthFacility().getUuid())
```

Comment:
alternatively I could just drop this check implies the tasks to be reassigned when a new officer working at the facility of the case is assigned? I think this might be even better?

Reply:
Not sure whether I understand you correctly, and I'm also not sure whether I properly understand the logic you've implemented. 

I think what I would do here is to keep the current logic that triggers `reassignTasks` to be called, and extend `reassignTasks` to only actually reassign tasks when the case's jurisdiction doesn't match the jurisdiction of the currently assigned user anymore. Since `reassignTasks` is also used to reassign tasks from users for which the user roles have changed, it would also be necessary to add a `forceReassignment` paramter to that method that is `false` if it's called from `onCaseChanged` and `true` if called from `UserFacadeEjb:removeUserAsSurveillanceAndContactOfficer`.

What do you think about that proposal?